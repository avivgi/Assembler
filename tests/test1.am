.define sz = 2
MAIN: mov r3, LIST[sz]
LOOP: jmp L1
prn #-5
mov STR[5], STR[2]
sub r1, r4
cmp r4, #sz
bne END
L1: bne LOOP
END: hlt
.define len = 4
STR: .string "abcdef"
LIST: .data 6, -9, len
cmp r6, #sz
cmp r7, #sz
cmp r6, #sz
cmp r7, #sz
K: .data 22
