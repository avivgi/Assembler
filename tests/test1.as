.define sz = 2
MAIN: mov r3, LIST[sz]
LOOP: jmp L1
mcr m_mcr1
cmp r4, #sz
bne END
endmcr
prn #-5
mov STR[5], STR[2]
sub r1, r4
m_mcr1
L1: bne LOOP
END: hlt
.define len = 4
mcr m_mcr2
cmp r6, #sz
cmp r7, #sz
endmcr
STR: .string "abcdef"
LIST: .data 6, -9, len
K: .data 22
m_mcr2
m_mcr2
K: .data 22
