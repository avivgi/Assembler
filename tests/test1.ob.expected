35 11
0100 00000000111000 ****!%*
0101 00000001100000 ***#%**
0102 00001000111010 **%*!%%
0103 00000000001000 *****%*
0104 00001001000100 **%#*#*
0105 00000111100010 **#!%*%
0106 00001100000000 **!****
0107 11111111101100 !!!!%!*
0108 00000000101000 ****%%*
0109 00001000011110 **%*#!%
0110 00000000010100 ****##*
0111 00001000011110 **%*#!%
0112 00000000001000 *****%*
0113 00000011111100 ***!!!*
0114 00000000110000 ****!**
0115 00000001110000 ***#!**
0116 00000010000000 ***%***
0117 00000000001000 *****%*
0118 00001010000100 **%%*#*
0119 00000111101010 **#!%%%
0120 00001010000100 **%%*#*
0121 00000110100010 **#%%*%
0122 00001111000000 **!!***
0123 00000001110000 ***#!**
0124 00000011000000 ***!***
0125 00000000001000 *****%*
0126 00000001110000 ***#!**
0127 00000011100000 ***!%**
0128 00000000001000 *****%*
0129 00000001110000 ***#!**
0130 00000011000000 ***!***
0131 00000000001000 *****%*
0132 00000001110000 ***#!**
0133 00000011100000 ***!%**
0134 00000000001000 *****%*
0135 00000001100001 ***#%*#
0136 00000001100010 ***#%*%
0137 00000001100011 ***#%*!
0138 00000001100100 ***#%#*
0139 00000001100101 ***#%##
0140 00000001100110 ***#%#%
0141 00000000000000 *******
0142 00000000000110 *****#%
0143 11111111110111 !!!!!#!
0144 00000000000100 *****#*
0145 00000000010110 ****##%
